Engine/TestStep.cs

```diff
@@ -955,6 +963,9 @@ internal static TestStepRun DoRun(this ITestStep Step, TestPlanRun planRun, Test
                         TapThread.Current.AbortToken);
                     try
                     {
+                        if (readInputsError != null)
+                            throw readInputsError;
```

Comment:
The exception stack trace is overwritten here. Is this intentional? If so, please add a comment. Otherwise, let's re-throw it instead.